tests/test_qaoa.py

```diff
@@ -764,3 +765,102 @@ def test_duplicates_remove_zeros(self):
         assert len(reduced_ops) == 1
         assert reduced_ops[0].name == "PauliZ"
         assert reduced_ops[0].wires.tolist() == [1]
+
+
+    def test_inner_net_flow_constraint_hamiltonian(self):
+        """Test if the _inner_net_flow_constraint_hamiltonian function returns the expected result on a manually-calculated
+        example of a 3-node complete digraph relative to the 0 node"""
+        g = nx.complete_graph(3).to_directed()
+        h = _inner_net_flow_constraint_hamiltonian(g, 0)
+
+        expected_ops = [
+            qml.Identity(0),
+            qml.PauliZ(0) @ qml.PauliZ(1),
+            qml.PauliZ(0) @ qml.PauliZ(2),
+            qml.PauliZ(0) @ qml.PauliZ(4),
+            qml.PauliZ(1) @ qml.PauliZ(2),
+            qml.PauliZ(1) @ qml.PauliZ(4),
+            qml.PauliZ(2) @ qml.PauliZ(4),
+        ]
+        expected_coeffs = [4, 2, -2, -2, -2, -2, 2]
+
+        assert expected_coeffs == h.coeffs
+        assert all([op.wires == exp.wires for op, exp in zip(h.ops, expected_ops)])
```

Comment:
I think we need another check to compare the ops.

Reply:
```suggestion
        assert expected_coeffs == h.coeffs
        assert all([op.wires == exp.wires for op, exp in zip(h.ops, expected_ops)])
        assert h.ops[0].name == "Identity"
        assert all("-".join(op1.name) == "-".join(op2.name) for op1, op2 in zip(h.ops[1:], expected_ops[1:]))
```
As an example - maybe there's a better way